packages/react/src/useView.ts

```diff
@@ -75,17 +111,26 @@ export function useView<VariablesT, F extends ViewFunction<VariablesT, any>>(
 
   const memoizedVariables = useStructuralMemo(variables);
   const memoizedOptions = useStructuralMemo(options);
-  const plan = useMemo(() => view.plan((memoizedVariables ?? {}) as unknown as VariablesT), [view, memoizedVariables]);
+  const [plan, dataPath] = useMemo((): [plan: GQLBuilderResult, dataPath: string[]] => {
+    if (typeof view == "string") {
+      return [{ query: inlineViewQuery, variables: { query: view, variables: memoizedVariables } }, ["gellyView"]];
+    } else {
+      return [view.plan((memoizedVariables ?? {}) as unknown as VariablesT), namespaceDataPath([view.gqlFieldName], view.namespace)];
+    }
+  }, [view, memoizedVariables]);
 
   const [rawResult, refresh] = useGadgetQuery(useQueryArgs(plan, memoizedOptions));
 
   const result = useMemo(() => {
-    const dataPath = namespaceDataPath([view.operationName], view.namespace);
     const data = get(rawResult.data, dataPath);
     const error = ErrorWrapper.errorIfDataAbsent(rawResult, dataPath, options?.pause);
 
     return { ...rawResult, data, error };
-  }, [view, options?.pause, rawResult]);
+  }, [dataPath, options?.pause, rawResult]);
 
   return [result, refresh];
 }
+
+const inlineViewQuery = `query InlineView($query: String!, $variables: JSONObject) { 
+  gellyView(query: $query, variables: $variables) 
+}`;
```

Comment:
this is a hardcoded query we make available in the GraphQLSchemaGenerator for running inline views, it is weird lookin but users never have to see it

Reply:
:+1: